### ISSUE TITLE
IAM/Agency with AK/SK

### ISSUE THREAD

**Issue by uli-max** [opentelekomcloud/terraform-provider-opentelekomcloud#2089]
# Terraform provider version
Terraform v1.3.4
on darwin_arm64
+ provider registry.terraform.io/opentelekomcloud/opentelekomcloud v1.33.0

# Affected Resource(s)
provider "opentelekomcloud" {
  agency_name        = var.agency_name
  agency_domain_name = var.agency_domain_name
  delegated_project  = var.delegated_project
  access_key         = var.access_key
  secret_key         = var.secret_key
  domain_name        = var.domain_name
  auth_url           = "https://iam.eu-de.otc.t-systems.com/v3"
}

# Terraform Configuration Files

Nothing special/other than in terraform provider documentaion.

# Debug Output/Panic Output

 Error: failed to authenticate:
│ error generating project client: Authentication Failed, error message: {"error":{"code":401,"message":"The X-Auth-Token is invalid!","title":"Unauthorized"}}
│ 
│   with provider["registry.terraform.io/opentelekomcloud/opentelekomcloud"],
│   on main.tf line 12, in provider "opentelekomcloud":
│   12: provider "opentelekomcloud" {

# Steps to Reproduce

Create an Agency for another domain.
Allow the other domain to access.  

# Expected Behavior

Using IAM Agency-Login with AK/SK for another Domain.  

# Actual Behavior

Error: failed to authenticate:
│ error generating project client: Authentication Failed, error message: {"error":{"code":401,"message":"The X-Auth-Token is invalid!","title":"Unauthorized"}}

# Important Factoids

Same code work with username/password

provider "opentelekomcloud" {
  agency_name        = var.agency_name
  agency_domain_name = var.agency_domain_name
  delegated_project  = var.delegated_project
  user_name          = var.user_name
  password           = var.password
  domain_name        = var.domain_name
  auth_url           = "https://iam.eu-de.otc.t-systems.com/v3"
}

# References

OTC Community Question:
https://community.open-telekom-cloud.com/community?id=community_question&sys_id=884c9addb7052510d15aa7b16b8c0213

**Comment by artem-lifshits** [comment]
Hello @uli-max . Looks like agency authorization via AK/SK is not available from OTC side (some headers that are being provided via AK/SK auth are not processed by the backend). 
It's possible to create a pretty complex workaround which would include 2 additional API requests and new logic, but this WA is not secure and wasn't approved.
Overall I recommend using password/login method.

**Comment by adesso-os** [comment]
The example code seems to match the [AKSK authentication configuration in the documentation](https://search.opentofu.org/provider/opentelekomcloud/opentelekomcloud/latest#assume-role). I have followed that documentation, and I receive the same error that was reported here originally.

What's going on here? If this is not supposed to work, why is it documented right on the first page of the provider?

It also seems like the background support for this was added years ago https://github.com/opentelekomcloud/gophertelekomcloud/commit/8f8f15118359b5a5fdc672901350634b6e9ffab5
